### PR TITLE
Refactor adyen to use go library

### PR DESCRIPTION
**Pull request by nirajjayantbolt**

### CODE REVIEW

File: gateways/adyen/adyen.go [hunk 1]
```diff
@@ -1,152 +1,92 @@
...
+	capture, err := modificationGateway.Capture(buildCaptureRequest(request, client.merchantAccount))
 	if err != nil {
-		return nil, err
+		return &sleet.CaptureResponse{Success: false, TransactionReference: ""}, err
 	}
-	convertedCode := strconv.Itoa(code)
-	return &sleet.CaptureResponse{Success: true, ErrorCode: &convertedCode}, nil
+	return &sleet.CaptureResponse{
+		Success:              true,
+		TransactionReference: capture.Response, // this might be wrong
```
Comment (jakechiang-bolt) [inline gateways/adyen/adyen.go:62]: I think this should be PspReference; https://docs.adyen.com/checkout/capture

File: gateways/adyen/adyen.go [hunk 2]
```diff
@@ -1,152 +1,92 @@
...
-
-	code, _, err := client.sendRequest("/refund", payload)
-	if err != nil {
-		return nil, err
-	}
-	convertedCode := strconv.Itoa(code)
-	return &sleet.RefundResponse{Success: true, ErrorCode: &convertedCode}, nil
+	return &sleet.RefundResponse{
+		Success:              true,
+		TransactionReference: refund.Response, // this might be wrong
```
Comment (jakechiang-bolt) [inline gateways/adyen/adyen.go:76]: Same here

File: gateways/adyen/adyen.go [hunk 3]
```diff
@@ -1,152 +1,92 @@
...
-
-	authorization := client.apiKey
-	req.Header.Add("X-API-key", authorization)
-	req.Header.Add("Content-Type", "application/json")
-	req.Header.Add("User-Agent", "sleet")
-
-	return req, nil
+	return &sleet.VoidResponse{
+		Success:              true,
+		TransactionReference: void.Response, // this might be wrong
```
Comment (jakechiang-bolt) [inline gateways/adyen/adyen.go:90]: Same here

**Comment by nirajjayantbolt** [comment]
Thanks yeah just finished updating test stuff - good to know that we need username and password for this

File: gateways/adyen/adyen.go (hunk 1 above)
Comment (nirajjayantbolt) [reply gateways/adyen/adyen.go:62]: Nice yeah let me update all of these

**Pull request merged**